lib/serial_terminal.pm

```diff
@@ -96,4 +101,105 @@ sub serial_term_prompt {
     return $serial_term_prompt;
 }
 
+
+=head2 download_file
+
+  download_file($src, $dst [, force => $force][, chunk_size => $cz][, chunk_retry => $cr])
+
+Download a file from worker to SUT using the current serial terminal.
+The file is split into chunks C<chunk_size> and each chunk is verified with
+checksum. If a chunk fail we retry it C<chunk_retry> times, before give up.
+To override destination us C<force>.
+This function die on any failure.
+=cut
+sub download_file {
+    my ($src, $dst, %opts) = @_;
+    $opts{chunk_size}  //= 1024 * 2;
+    $opts{chunk_retry} //= 16;
```

Comment:
any reason why you came up with 16 retries? have you encountered problems while testing this of a failed chunk?

Reply:
No reason...